### ISSUE TITLE
Opencv Problem when porting to Android

### ISSUE THREAD

**Issue by xiaodian2234** [HKUST-Aerial-Robotics/VINS-Mobile#83]
I have porting VINS to Android but I find that the performance of opencv on Android is much lower than IOS. 
Now I have difficulty in using calcOpticalFlowPyrLK on Android, which cost around 12ms on Android but less than 5ms on IOS.   I think this is probably because multithreading on Android is not high-performancly supported as on IOS. 
Could you suggest some method or other library that can realize Optical Flow on Android? Thanks a lot.

**Comment by sandmen** [comment]
@xiaodian2234,
on Android, you have very different results from device to device. :-(
Apple as the advantage, to support only "one" HW.
Do you have anywhere your port of VINS, so, I can check it on my device... ?

**Comment by StevenCui** [comment]
Hi，i am porting to android too, and i am testing the accuracy of android project with MH-05 dataset. Do u have tested it with dataset?

**Comment by xiaodian2234** [comment]
@sandmen I have tested on different Android phones. It seems that higher hardware configuration can get better result.

**Comment by xiaodian2234** [comment]
@StevenCui No. I only test it online or by recording data in our environment.

**Comment by TouqeerAhmad** [comment]
@xiaodian2234 Hi, do you have any plans to make the android version public ??

**Comment by shopkauf** [comment]
@xiaodian2234 : would you mind to share your android version with me? Thank you!

**Comment by FangGet** [comment]
opencv has an 3rdpary named carotene where opticalflow calculation is conducted on NEON, just have a try with it @xiaodian2234

**Comment by xiaodian2234** [comment]
@FangGet I have tested it. The average time of LK has reduced about 5ms. But the main problem in our Android phone is that sometimes the course can jump to about 20ms . The procedure seems not stable on current platform. We think this may be caused by cpu schedule.

**Comment by bileki** [comment]
@xiaodian2234 your android version is online? I would love to test on my motoG1 to check the performance as I am getting the data from motoG1 and doing the calculations on a server.
